File: tests/smoke/bootstrap_smoke.py
"""Phase 0 smoke gate, Python edition."""

import asyncio
import os
import sys
import time
from pathlib import Path

ROOT = Path(__file__).resolve().parents[2]
sys.path.insert(0, str(ROOT))

from dotenv import load_dotenv

load_dotenv(ROOT / ".env.local")
load_dotenv(ROOT / ".env")

from server.agent.bootstrap import run_bootstrap  # noqa: E402

PROMPTS = [
    "I'm training for the Boston marathon in October, I run 4 times a week",
    "Plan my Japan honeymoon, two weeks, Tokyo and Kyoto",
    "Track my software engineering job applications",
]


async def main() -> int:
    if "GEMINI_API_KEY" not in os.environ:
        print("GEMINI_API_KEY missing")
        return 1

    model = os.environ.get("GEMINI_MODEL", "gemini-3-flash-preview")
    print(f"Using model: {model}")

    passed = 0
    for p in PROMPTS:
        print(f'\n→ Prompt: "{p}"')
        t0 = time.time()
        try:
            r = await run_bootstrap(p)
        except Exception as e:
            print(f"❌ Threw: {type(e).__name__}: {e}")
            continue
        dt = int((time.time() - t0) * 1000)
        tools = ", ".join(t.name for t in r.tools)
        print(f"  ✅ name={r.name} icon={r.icon} appType={r.appType} ({dt}ms)")
        print(
            f"     root widget={r.tree.type} tools=[{tools}] data keys=[{', '.join(r.data.keys())}]"
        )
        passed += 1

    print(f"\n{passed}/{len(PROMPTS)} prompts passed")
    return 0 if passed == len(PROMPTS) else 2


if __name__ == "__main__":
    sys.exit(asyncio.run(main()))

File: tests/smoke/bootstrap-smoke.ts
import * as dotenv from "dotenv";
// Load .env.local first (preferred), fall back to .env
dotenv.config({ path: ".env.local" });
dotenv.config();
import { GoogleGenerativeAI } from "@google/generative-ai";
import { bootstrapResponseSchema } from "../../server/agent/schema.js";
import { BOOTSTRAP_SYSTEM } from "../../server/agent/prompts.js";

const PROMPTS = [
  "I'm training for the Boston marathon in October, I run 4 times a week",
  "Plan my Japan honeymoon, two weeks, Tokyo and Kyoto",
  "Track my software engineering job applications",
];

async function runOne(client: GoogleGenerativeAI, modelName: string, message: string): Promise<boolean> {
  console.log(`\n→ Prompt: "${message}"`);
  const t0 = Date.now();

  const model = client.getGenerativeModel({
    model: modelName,
    systemInstruction: BOOTSTRAP_SYSTEM,
    generationConfig: {
      responseMimeType: "application/json",
    },
  });

  const result = await model.generateContent(message);
  const text = result.response.text();
  const elapsed = Date.now() - t0;
  console.log(`← ${text.length} chars in ${elapsed}ms`);

  let parsed: unknown;
  try {
    parsed = JSON.parse(text);
  } catch (e) {
    console.error("❌ JSON parse failed");
    console.error(text.slice(0, 500));
    return false;
  }

  const v = bootstrapResponseSchema.safeParse(parsed);
  if (!v.success) {
    console.error("❌ Schema validation failed:");
    console.error(JSON.stringify(v.error.flatten(), null, 2));
    console.error("Response was:", text.slice(0, 500));
    return false;
  }

  const r = v.data;
  console.log(`  ✅ name=${r.name} icon=${r.icon} appType=${r.appType}`);
  console.log(`     root widget=${(r.tree as any).type} tools=[${r.tools.map((t) => t.name).join(", ")}] data keys=[${Object.keys(r.data).join(", ")}]`);
  return true;
}

async function main() {
  const apiKey = process.env.GEMINI_API_KEY;
  if (!apiKey) {
    console.error("GEMINI_API_KEY not set");
    process.exit(1);
  }
  const modelName = process.env.GEMINI_MODEL || "gemini-3-pro-preview";
  console.log(`Using model: ${modelName}`);

  const client = new GoogleGenerativeAI(apiKey);

  const results: boolean[] = [];
  for (const p of PROMPTS) {
    try {
      results.push(await runOne(client, modelName, p));
    } catch (e) {
      console.error("❌ Threw:", (e as Error).message);
      results.push(false);
    }
  }

  const passed = results.filter(Boolean).length;
  console.log(`\n${passed}/${results.length} prompts passed`);
  process.exit(passed === results.length ? 0 : 2);
}

main();
